src/OnboardingSPA/components/SocialMediaForm/index.js

```diff
@@ -1,235 +1,308 @@
-                    </label>
-                    <Tooltip content={activeError.includes(SocialMediaSites[social]) ? showErrorMessage(SocialMediaSites[social]) : 'hide'} direction="top">
-                        <input className={`${activeError.includes(SocialMediaSites[social]) ? "social-form__box-error" : "social-form__box"}`} type="url" id={`${SocialMediaSites[social]}`} value={SocialMediaStates[social]} onChange={(value) => { handleChange(value) }} />
-                    </Tooltip>
-                </div>
-            );
-        }
-        return socialBoxes;
-    }
-
-    return (
-        <div className="social-form">
-            <div className="social-form__top-row" onClick={(e) => { handleAccordion(e)}}>
-                <div className="social-form__top-row_heading">
-                    {__(
-                        "Social Media",
-                        'wp-module-onboarding'
-                    )}
-                </div>
-                <div className={`social-form__top-row_icon ${isSocialFormOpen ? 'social-form__top-row_icon_opened' : ''}`}></div>
-            </div>
-            <form className={isSocialFormOpen ? 'social-form__main-active' : 'social-form__main-hidden'} onSubmit={(e) => { handleSubmit(e) }}>
-                {buildSocialBoxes()}
-            </form>
-        </div>
-    );
+		) {
+			const otherURLS = socialData.other_social_urls;
+			if ( Object.keys( otherURLS ).includes( 'yelp_url' ) )
+				setYelp( otherURLS.yelp_url ?? '' );
+
+			if ( Object.keys( otherURLS ).includes( 'tiktok_url' ) )
+				setTikTok( otherURLS.tiktok_url ?? '' );
+		}
+	}, [ socialData ] );
+
+	const handleAccordion = () => {
+		setIsSocialFormOpen( ! isSocialFormOpen );
+	};
+
+	const navigationCallback = ( callback ) => {
+		setShowModal( true );
+		setModalCallback( () => callback );
+	};
+
+	const saveData = ( value, triggerID ) => {
+		switch ( triggerID ) {
+			case SocialMediaSites.FACEBOOK:
+				setFacebook( value );
+				socialMediaDB.facebook_site = value;
+				break;
+			case SocialMediaSites.TWITTER:
+				setTwitter( value );
+				socialMediaDB.twitter_site = value;
+				break;
+			case SocialMediaSites.INSTAGRAM:
+				setInstagram( value );
+				socialMediaDB.instagram_url = value;
+				break;
+			case SocialMediaSites.YOUTUBE:
+				setYouTube( value );
+				socialMediaDB.youtube_url = value;
+				break;
+			case SocialMediaSites.LINKEDIN:
+				setLinkedIn( value );
+				socialMediaDB.linkedin_url = value;
+				break;
+			case SocialMediaSites.YELP:
+				setYelp( value );
+				socialMediaDB.other_social_urls.yelp_url = value;
+				break;
+			case SocialMediaSites.TIKTOK:
+				setTikTok( value );
+				socialMediaDB.other_social_urls.tiktok_url = value;
+				break;
+		}
+		setSocialData( socialMediaDB );
+	};
+
+	const getErrorType = () => {
+		let error = ERROR_TYPES.NONE;
+		for ( const key in activeErrorTypes ) {
+			if ( activeErrorTypes[ key ] !== ERROR_TYPES.NONE )
+				error = activeErrorTypes[ key ];
```

Comment:
So that was done assuming that the later filled fields that are Tiktok and yelp would have been filled more recently as compared to Facebook and say both have errors in that case the later filled error should be shown rather than the old one. so the error gets replaced with a newer error.